### PR TITLE
HIVE-25631: Initiator speed-up: only read compaction history once per…

### PR DESCRIPTION
**Pull request by deniskuzZ**
… loop

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java [hunk 1]
```diff
@@ -474,9 +508,7 @@ private boolean isEligibleForCompaction(CompactionInfo ci,
       // as it may be time to compact again even though we haven't cleaned.
       // todo: this is not robust. You can easily run `alter table` to start a compaction between
       // the time currentCompactions is generated and now
```
Comment (klcopp) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java:510]: Maybe add another comment here about what "failed compactions" means

**Review by klcopp — COMMENTED** [review]
Looks good, a couple small suggestions

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java (hunk 1 above)
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java]: done

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java [hunk 2]
```diff
@@ -508,14 +540,6 @@ private boolean isEligibleForCompaction(CompactionInfo ci,
         return false;
       }
 
-      if (txnHandler.checkFailedCompactions(ci)) {
```
[the remark this answers is not in the document]
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java:511]: It's a public API, could be used by an external application, so not safe to remove.

**Review by klcopp — APPROVED** [review]

**Pull request merged**